Library/Homebrew/cmd/which-formula.sh

```diff
@@ -53,6 +53,7 @@ download_and_cache_executables_file() {
       max_time=10
       retries=0
     fi
+    mkdir -p "${HOMEBREW_CACHE}/api/internal"
```

Comment:
Would be nice to rely on the `DATABASE_FILE` variable instead to avoid the future possibility of inconsistency when this is updated.

```suggestion
    mkdir -p "${DATABASE_FILE%/*}"
```